src/Ng/Jobs/MonitoringProcessorJob.cs

```diff
@@ -172,14 +174,33 @@ private async Task ProcessPackagesAsync(CancellationToken token)
                     Logger.LogCritical(
                         NuGet.Services.Metadata.Catalog.Monitoring.LogEvents.QueueMessageFatalFailure,
                         aggregateException,
-                        "Failed to process queue message");
+                        "Failed to process queue message.");
                 }
             }
 
             // If we failed to run validations and failed to save the failed validation, we cannot remove the message from the queue.
             if (messageWasProcessed)
             {
-                await _queue.RemoveAsync(queueMessage, token);
+                try
+                {
+                    await _queue.RemoveAsync(queueMessage, token);
+                }
+                catch (StorageException storageException)
+                {
+                    if (storageException.RequestInformation.ExtendedErrorInformation.ErrorCode == QueueErrorCodeStrings.MessageNotFound)
+                    {
+                        Logger.LogWarning(
+                            "Queue message for {PackageId} {PackageVersion} no longer exists. Message was likely handled by another instance of the job.",
```

Comment:
I would consider including the exception just in case we want to look deeper into this case in the future.

Reply:
added